pysph/sph/acceleration_eval_opencl_helper.py

```diff
@@ -81,9 +82,15 @@ def _call_kernel(self, info, extra_args):
                 cache._start_idx_gpu.array.data,
                 cache._neighbors_gpu.array.data
             ] + extra_args
-            call(*args)
+            event = call(*args)
+            if self._profile:
+                profile(call.get_info(cl.kernel_info.FUNCTION_NAME),
+                        event)
         else:
-            call(*(args + extra_args))
+            event = call(*(args + extra_args))
+            if self._profile:
```

Comment:
Again, if profile was a decorator or so we could either do the profiling or just return the original if not profiling.